src/sentry/tasks/merge.py

```diff
@@ -99,99 +112,121 @@ def merge_group(
             }
         )
 
-    model_list = tuple(EXTRA_MERGE_MODELS) + (
-        Activity, GroupAssignee, GroupEnvironment, GroupHash, GroupRuleStatus,
-        GroupSubscription, EventMapping, Event, UserReport, GroupRedirect,
-        GroupMeta,
-    )
-
-    has_more = merge_objects(
-        model_list,
-        group,
-        new_group,
-        logger=logger,
-        transaction_id=transaction_id,
-    )
+    try:
+        group = Group.objects.get(id=from_object_id)
+    except Group.DoesNotExist:
+        from_object_ids.remove(from_object_id)
 
-    if has_more:
-        merge_group.delay(
-            from_object_id=from_object_id,
-            to_object_id=to_object_id,
-            transaction_id=transaction_id,
-            recursed=True,
-            eventstream_state=eventstream_state,
+        logger.warn(
+            'group.malformed.invalid_id',
+            extra={
+                'transaction_id': transaction_id,
+                'old_object_id': from_object_id,
+            }
         )
-        return
-
-    features.merge(new_group, [group], allow_unsafe=True)
-
-    environment_ids = list(
-        Environment.objects.filter(
-            projects=group.project
-        ).values_list('id', flat=True)
-    )
-
-    for model in [tsdb.models.group]:
-        tsdb.merge(
-            model,
-            new_group.id,
-            [group.id],
-            environment_ids=environment_ids if model in tsdb.models_with_environment_support else None
+    else:
```

Comment:
This huge else block (rather than an early return) seems a little weird, but I didn't put any thought into how it could otherwise be structured.

Reply:
Yeah, it's annoying and ugly but is there to deal with the delay/eventstream at the end which would have to be duplicated otherwise I think.